components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2492,6 +2507,27 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
         }
     }
 
+    /**
+     * Return whether the request of updating the tokenEndpointAllowReusePvtKeyJwt is valid.
+     *
+     * @param tokenEndpointAuthMethod     token endpoint client authentication method.
+     * @param tokenEndpointAllowReusePvtKeyJwt During client authentication whether to reuse private key JWT.
+     * @return True if tokenEndpointAuthMethod and tokenEndpointAllowReusePvtKeyJwt is NOT in the correct format.
+     */
+    private boolean isInvalidTokenEPReusePvtKeyJwtRequest(String tokenEndpointAuthMethod,
+                                                          Boolean tokenEndpointAllowReusePvtKeyJwt) {
+
+        if (StringUtils.isNotBlank(tokenEndpointAuthMethod)) {
+            if (tokenEndpointAuthMethod.equals(PRIVATE_KEY_JWT)) {
+                return tokenEndpointAllowReusePvtKeyJwt == null;
+            } else {
+                return tokenEndpointAllowReusePvtKeyJwt != null;
+            }
+        } else {
```

Comment:
we don't need this else block.
```
    if (StringUtils.isBlank(tokenEndpointAuthMethod)) {
        return tokenEndpointAllowReusePvtKeyJwt != null;
    }

    if (tokenEndpointAuthMethod.equals(PRIVATE_KEY_JWT)) {
        return tokenEndpointAllowReusePvtKeyJwt == null;
    }

    return tokenEndpointAllowReusePvtKeyJwt != null;
```

Reply:
Addressed with 1346c762ac6987102496807d3d874923edeb3003